### PR TITLE
Fix analytics for App Engine deploy

### PR DESCRIPTION
**Pull request by chanseokoh**
We were always sending "standard" deploy events even when deploying to flex.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java [hunk 1]
```diff
@@ -140,10 +138,17 @@ private static boolean checkProjectErrors(IProject project) throws CoreException
     return severity != IMarker.SEVERITY_ERROR;
   }
 
+  protected void onDeployJobCreation() {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java]: Could we make this method name less about the implementation? `onDeployStart()`?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java [hunk 2]
```diff
@@ -140,10 +138,17 @@ private static boolean checkProjectErrors(IProject project) throws CoreException
...
+    // Allow subclasses to perform actions before launching the deploy job.
+  }
+
+  protected void onSuccessfulDeploy() {
```
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java]: Mirror the `onDeploy*` event name style: `onDeploySuccess()`?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java (hunk 1 above)
Comment (chanseokoh) [reply plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java]: Makes sense.

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java [hunk 3]
```diff
@@ -140,10 +138,17 @@ private static boolean checkProjectErrors(IProject project) throws CoreException
     return severity != IMarker.SEVERITY_ERROR;
   }
 
+  protected void onDeployStart() {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java]: should be abstract?

File: plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java [hunk 4]
```diff
@@ -140,10 +138,17 @@ private static boolean checkProjectErrors(IProject project) throws CoreException
...
+    // Allow subclasses to perform actions before starting deploy.
+  }
+
+  protected void onDeploySuccess() {
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployCommandHandler.java]: abstract

**Comment by chanseokoh** [comment]
@elharo had another idea without method overriding, which I think is actually better. Updated.

**Pull request merged**